### PR TITLE
Issue 32, ejs templates using a not existing include or a page attribute missing

### PR DESCRIPTION
**Pull request by smartiniOnGitHub**
Hi all, this is the implementation of the fix.
Due to lack of time now it's without tests but I'll add them in a few days.
I published now so maybe in the meantime someone could do some initial review and feedback.
Thanks for now. Bye

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -49,12 +49,21 @@ function fastifyView (fastify, opts, next) {
         return
       }
 
-      lru.set(page, engine.compile(html, options))
+      try {
+        lru.set(page, engine.compile(html, options))
+      } catch (error) {
+        that.send(error)
+        return
+      }
```
Comment (mcollina) [inline index.js]: Can you split `lru.set()` and `engine.compile`? I would like not to catch errors from `lru.set()`.

File: index.js [hunk 2]
```diff
@@ -49,12 +49,21 @@ function fastifyView (fastify, opts, next) {
...
 
       if (!that.res.getHeader('content-type')) {
         that.header('Content-Type', 'text/html')
       }
-      that.send(lru.get(page)(data))
+      try {
+        that.send(lru.get(page)(data))
+      } catch (error) {
+        that.send(error)
+      }
```
Comment (mcollina) [inline index.js]: can you just wrap calling the function? The rest should not be wrapped.

File: index.js (hunk 1 above)
Comment (smartiniOnGitHub) [reply index.js]: ok, in next push I'll do that change

File: index.js (hunk 2 above)
Comment (smartiniOnGitHub) [reply index.js]: I tried but without that inside a try/catch I have the same a crash of the server ...
So I could update even this block in a similar way of the previous change.

**Comment by smartiniOnGitHub** [comment]
@mcollina Hi Matteo, all changes (even tests) should be done now ... tell me if there is something to change, or if it's good enough to merge (I hope :-) ). Thanks again. Bye

**Comment by mcollina** [comment]
@delvedor are you ok with this?

**Comment by delvedor** [comment]
Sorry I missed this!
I'm 👍 !

**Pull request merged**